bitset.go

```diff
@@ -530,9 +530,11 @@ func (b *BitSet) NextClear(i uint) (uint, bool) {
 		return 0, false
 	}
 	for x < len(b.set) {
-		index = uint(x)*wordSize + trailingZeroes64(^b.set[x])
-		if b.set[x] != allBits && index < b.length {
-			return index, true
+		if b.set[x] != allBits {
```

Comment:
I don't mind this change, but I don't see how this is an optimization. And I don't see how it relates to bound checks. Can you explain?

Reply:
this was writing to `index` in every iteration, you can see from the benchstat numbers that the baseline of NextSet was lower than NextClear, and it was because of this (now they're the same).

Reply:
In the case where `b.set[x] == allBits` is expected to be true, then this new code can be faster.